File: app/src/main/java/edu/learn/newsreader/ViewModel/NewsViewModel.java
package edu.learn.newsreader.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import edu.learn.newsreader.Modals.Article;
import edu.learn.newsreader.Repository.NewsRepos;

public class NewsViewModel extends AndroidViewModel {
    private LiveData<List<Article>> newsLiveData;
    private NewsRepos newsRepos;
    public NewsViewModel(@NonNull Application application) {
        super(application);
        newsRepos = new NewsRepos(application);
        newsLiveData = newsRepos.getAllNews();
    }

    public void startApiCall(String country, String api_key){
        newsRepos.getNewsFromApi(country,api_key);
    }

    public LiveData<List<Article>> getAllNews(){
        return newsLiveData;
    }
}

File: app/src/main/java/edu/learn/newsreader/viewModel/NewsViewModel.kt
package edu.learn.newsreader.viewModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import edu.learn.newsreader.modals.Article
import edu.learn.newsreader.repository.NewsRepos

class NewsViewModel(application: Application) : AndroidViewModel(application) {
    val allNews: LiveData<List<Article>>
    private val newsRepos = NewsRepos(application)

    init {
        allNews = newsRepos.allNews
    }

    fun startApiCall(country: String?, apiKey: String?, swipeRefreshLayout: SwipeRefreshLayout?) {
        newsRepos.getNewsFromApi(country, apiKey, swipeRefreshLayout!!)
    }
}
